sippy-ng/src/component_readiness/CompReadyUtils.js

```diff
@@ -501,6 +501,41 @@ export function validateData(data) {
   return ['']
 }
 
+export function groupIncidentIncidents(incidents, groupedIncidents) {
+  incidents.forEach((incident) => {
+    if (incident.incidents && incident.incidents.length > 0) {
+      groupIncidents(incident.incidents, groupedIncidents, incident)
+    }
+  })
+}
+
+export function groupIncidents(incidents, groupedIncidents, detailsIncident) {
+  incidents.forEach((i) => {
+    if (!groupedIncidents.has(i.incident_group_id)) {
+      groupedIncidents.set(i.incident_group_id, {
+        issue: i.issue,
+        incidents: [],
+      })
+    }
+    let incidentsGroup = groupedIncidents.get(i.incident_group_id)
+    i.details = detailsIncident
+    incidentsGroup.incidents = incidentsGroup.incidents.concat(i)
+  })
+}
+
+export function createGroupedIncidentArray(groupedIncidents) {
+  return Array.from(groupedIncidents, ([group_id, grouped_incidents]) => ({
+    group_id,
+    grouped_incidents,
+  }))
+}
+
+export function mergeIncidents(incidents, detailsIncident) {
+  let groupedIncidents = new Map()
+  groupIncidents(incidents, groupedIncidents, detailsIncident)
+  return createGroupedIncidentArray(groupedIncidents)
+}
```

Comment:
I could use some comments on these functions, a little hard to put the picture together of what's going on.